src/main/java/mandel/gameoflife/GridComponent.java

```diff
@@ -0,0 +1,49 @@
+package mandel.gameoflife;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+public class GridComponent extends JComponent {
+
+    private final Grid grid;
+    private final int cellSize = 20;
+
+    public GridComponent(Grid grid) {
+        this.grid = grid;
+        addMouseListener(new MouseAdapter() {
+            @Override
+            public void mouseClicked(MouseEvent e) {
+                int x = e.getX() / cellSize;
+                int y = e.getY() / cellSize;
+
+                if (grid.isAlive(x, y)) {
+                    grid.remove(x, y);
+                } else {
+                    grid.put(x, y);
+                }
+                grid.setInitial();
```

Comment:
It saves the initial setup of the grid after the user clicks on the squares. When the reset button is pressed, the reset() method is called (in GridFrame) and it can access the original grid.